pallets/creditcoin/src/lib.rs

```diff
@@ -469,6 +470,15 @@ pub mod pallet {
 
 		/// The external address is malformed or otherwise invalid for the platform.
 		MalformedExternalAddress,
+
+		/// Unsuccessful publick key recovery from a message and an ecdsa signature.
+		EcdsaRecoveryFailed,
```

Comment:
A higher-level error name would be better, users might not know what ecdsa recovery is or what it means. Since the error can only occur if the signature is bad, maybe something like `InvalidSignature` or `InvalidAddressSignature`?